compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -285,6 +288,33 @@ impl SelfProfilerRef {
         })
     }
 
+    /// Record the size of an artifact that the compiler produces
+    ///
+    /// `artifact_kind` is the class of artifact (e.g., query_cache, object_file, etc.)
+    /// `artifact_name` is an identifier to the specific artifact being stored (usually a filename)
+    #[inline(always)]
+    pub fn artifact_size<A>(&self, artifact_kind: &str, artifact_name: A, size: u64)
```

Comment:
Why inline this?

Reply:
All of these event recording methods are `inline(always)` because they are thin wrappers around a function call that is never inlined. See https://github.com/rust-lang/rust/pull/87404/files#diff-98f589d77d5a9fcfd42c01231da93e4b0604804c976923f0ba3a676bc6c07d79L202-L206.

So their code results in something like `if some_condition { call_function_that_is_never_inlined() }`. 

For `artifact_size` in particular it might not be so important because it isn't called very often. But I personally preferred it if all of these similar methods followed the same pattern.